apps/src/music/MusicView.jsx

```diff
@@ -0,0 +1,633 @@
+/** @file Top-level view for Music */
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import {Provider, connect} from 'react-redux';
+import CustomMarshalingInterpreter from '../lib/tools/jsinterpreter/CustomMarshalingInterpreter';
+import {parseElement as parseXmlElement} from '../xml';
+import queryString from 'query-string';
+import {baseToolbox, createMusicToolbox} from './blockly/toolbox';
+import Instructions from './Instructions';
+import SharePlaceholder from './SharePlaceholder';
+import Controls from './Controls';
+import Timeline from './Timeline';
+import {MUSIC_BLOCKS} from './blockly/musicBlocks';
+import {BlockTypes} from './blockly/blockTypes';
+import MusicPlayer from './player/MusicPlayer';
+import InputContext from './InputContext';
+import {Triggers} from './constants';
+import {musicLabDarkTheme} from './blockly/themes';
+import AnalyticsReporter from './analytics/AnalyticsReporter';
+import {getStore} from '@cdo/apps/redux';
+import {SignInState} from '@cdo/apps/templates/currentUserRedux';
+
+function getRandomIntInclusive(min, max) {
+  min = Math.ceil(min);
+  max = Math.floor(max);
+
+  // The maximum is inclusive and the minimum is inclusive.
+  return Math.floor(Math.random() * (max - min + 1) + min);
+}
+
+const baseUrl = 'https://cdo-dev-music-prototype.s3.amazonaws.com/';
+
+var hooks = {};
+
+class UnconnectedMusicView extends React.Component {
+  static propTypes = {
```

Comment:
is this a class component because of the need for redux?

Reply:
Yeah, this just started out as a class component but with the need for Redux and the fact that it currently houses a lot of the blockly logic, it made sense to keep it a class. As we move stuff out of here it would be ideal to convert this (and the other views) into a functional component.